src/game/client/components/console.cpp

```diff
@@ -151,6 +153,13 @@ void CGameConsole::CInstance::OnInput(IInput::CEvent Event)
 					char *pEntry = m_History.Allocate(m_Input.GetLength()+1);
 					mem_copy(pEntry, m_Input.GetString(), m_Input.GetLength()+1);
 				}
+				// if the user isn't authenticating or disconnected, show their upcoming commands in the scrollback
+				if ((str_comp(pConPrompt, "ENTER PASSWORD> ") != 0) && (str_comp(pConPrompt, "NOT CONNECTED> ") != 0))
+				{
+					char aBuf[128];
+					str_format(aBuf, sizeof(aBuf), "> %s", m_Input.GetString());
+					PrintLine(aBuf, false);
```

Comment:
You could add this to body of the existing `if(m_Type == CONSOLETYPE_LOCAL || m_pGameConsole->Client()->RconAuthed())` a few lines above, as this checks exactly what you are trying to achieve. Then you don't need the `pConPrompt` variable and the string comparisons.

Reply:
Oh thanks!